lxd/apparmor/rsync.go

```diff
@@ -0,0 +1,221 @@
+//go:build linux && cgo && !agent
+
+package apparmor
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"text/template"
+
+	"github.com/lxc/lxd/lxd/sys"
+	"github.com/lxc/lxd/shared"
+	"github.com/lxc/lxd/shared/subprocess"
+)
+
+var rsyncProfileTpl = template.Must(template.New("rsyncProfile").Parse(`#include <tunables/global>
+profile "{{ .name }}" flags=(attach_disconnected,mediate_deleted) {
+  #include <abstractions/base>
+
+  capability dac_override,
+  capability dac_read_search,
+  capability mknod,
+  capability chown,
+  capability fsetid,
+  capability fowner,
+
+  unix, # TODO Perhaps a stricter restriction can be made here
+  
+  @{PROC}/@{pid}/cmdline r,
+  @{PROC}/@{pid}/cpuset r,
+  {{ .rootPath }}/{etc,lib,usr/lib}/os-release r,
+
+  {{ .logPath }}/*/netcat.log rw,
+
+  {{ .rootPath }}/run/{resolvconf,NetworkManager,systemd/resolve,connman,netconfig}/resolv.conf r,
+  {{ .rootPath }}/run/systemd/resolve/stub-resolv.conf r,
+
+{{range $index, $element := .allowedCmdPaths}}
+  {{$element}} mixr,
+{{- end }}
+
+  {{ .sourcePath }}/** r,
+  {{ .sourcePath }}/ r,
+
+
+{{- if .dstPath }}
+  {{ .dstPath }}/** rw,
+  {{ .dstPath }}/ rw,
+{{- end }}
+
+{{- if .snap }}
+  # Snap-specific libraries
+  /snap/lxd/*/lib/**.so* mr,
+
+  /var/snap/lxd/common/lxd.debug mixr,
```

Comment:
I don't understand why `rsync` would need to execute `lxd.debug`?

Reply:
In function `sendSetup` in https://github.com/lxc/lxd/blob/master/lxd/rsync/rsync.go#L124 variable `execPath` has a value of current lxd executable. It is got from https://github.com/lxc/lxd/blob/master/lxd/instance/drivers/driver_lxc.go#L5466 (last argument).